eng/pipelines/templates/jobs/archetype-go-release.yml

```diff
@@ -4,67 +4,86 @@ parameters:
   ServiceDirectory: ''
 
 stages:
-  - ${{if and(eq(variables['Build.Reason'], 'Manual'), eq(variables['System.TeamProject'], 'internal'))}}:
-    - stage: Release
-      displayName: 'Release: ${{ parameters.ServiceDirectory }}'
-      dependsOn: ${{ parameters.DependsOn }}
-      condition: and(succeeded(), ne(variables['SetDevVersion'], 'true'), ne(variables['Skip.Release'], 'true'), ne(variables['Build.Repository.Name'], 'Azure/azure-sdk-for-go-pr'))
-      jobs:
-        - deployment: TagRepository
-          displayName: "Create release tag"
-          condition: and(succeeded(), ne(variables['Skip.TagRepository'], 'true'))
-          environment: github
+  - stage: CheckRelease
+    displayName: 'Check Release: ${{ parameters.ServiceDirectory }}'
+    dependsOn: ${{ parameters.DependsOn }}
+    condition: and(succeeded(), ne(variables['SetDevVersion'], 'true'), ne(variables['Skip.Release'], 'true'), ne(variables['Build.Repository.Name'], 'Azure/azure-sdk-for-go-pr'))
+    jobs:
+      - job: CheckReleaseJob
+        displayName: "Check whether need to release"
+        steps:
+          - task: PowerShell@2
+            name: Verify
+            inputs:
+              filePath: ./eng/scripts/Verify-NeedToRelease.ps1
+              arguments: >
+                -PackageName 'sdk/${{ parameters.ServiceDirectory }}'
+                -ServiceDirectory '${{ parameters.ServiceDirectory }}'
+                -repoId Azure/azure-sdk-for-go
+                -workingDirectory $(System.DefaultWorkingDirectory)
+              pwsh: true
+            env:
+              GH_TOKEN: $(azuresdk-github-pat)
+  - stage: Release
+    displayName: 'Release: ${{ parameters.ServiceDirectory }}'
+    dependsOn: CheckRelease
+    condition: and(succeeded(), eq(dependencies.CheckRelease.outputs['CheckReleaseJob.Verify.NeedToRelease'], 'true'))
+    jobs: 
+      - deployment: TagRepository
+        displayName: "Create release tag"
+        condition: and(succeeded(), eq(variables['NeedToRelease'], 'true'), ne(variables['Skip.TagRepository'], 'true'))
```

Comment:
Based on https://docs.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch#use-outputs-in-a-different-stage

I suspect you need to get this value from the outputs similar to the other usage. Actually, do we need the condition here as well as on the stage? Perhaps this one can be removed. 

I think this is likely the reason for the issue that @seankane-msft is seeing.

Reply:
Yes, this job condition is wrong and also redundant. I've removed it by this [PR](https://github.com/Azure/azure-sdk-for-go/pull/16355). Thanks @weshaggard @seankane-msft for your validation and correction.